src/SignalR/server/Core/src/DefaultHubLifetimeManager.cs

```diff
@@ -271,14 +292,22 @@ public override Task SendUserAsync(string userId, string methodName, object?[] a
     public override Task OnConnectedAsync(HubConnectionContext connection)
     {
         _connections.Add(connection);
+        connection.Features.Set(new GroupTrackerFeature()); //add a group tracker to every concection
         return Task.CompletedTask;
     }
 
     /// <inheritdoc />
     public override Task OnDisconnectedAsync(HubConnectionContext connection)
     {
+        //now remove from tracked groups one by one
+        //this is faster than calling _groups.RemoveDisconnectedConnection
+        //because that method iteratas through ALL the groups
```

Comment:
Remove, references a function that won't exist anymore and is mostly self-explanatory